libkbfs/block_retrieval_queue.go

```diff
@@ -276,17 +282,22 @@ func (brq *blockRetrievalQueue) Shutdown() {
 }
 
 // TogglePrefetcher allows upstream components to turn the prefetcher on or off
-func (brq *blockRetrievalQueue) TogglePrefetcher(ctx context.Context, enable bool) error {
-	brq.prefetchMtx.Lock()
-	defer brq.prefetchMtx.Unlock()
+func (brq *blockRetrievalQueue) TogglePrefetcher(ctx context.Context, enable bool) (err error) {
+	var shutdownCh <-chan struct{}
+	func() {
+		brq.prefetchMtx.Lock()
+		defer brq.prefetchMtx.Unlock()
+		shutdownCh = brq.prefetcher.Shutdown()
```

Comment:
Should the prefetcher be set to `nil` here while under lock?  Otherwise the other users of it might try to use a shutdown prefetcher before it is replaced.  (Also, if `!enable`, the prefetcher is never set to `nil`.)